### PR TITLE
Initial pass at Raspberry Pi support.

### PR DESCRIPTION
**Pull request by jshuping**
- All the tags/roles work now
- Started a small 'rpi' role
- resolves pigmonkey/spark#51

### CODE REVIEW

File: roles/base/tasks/packages.yml [hunk 1]
```diff
@@ -26,7 +27,7 @@
     - packages
 
 - name: Install cower
-  aur: name=cower user={{ user.name }} skip_pgp=yes
+  pacman: name=cower state=present
```
Comment (pigmonkey) [inline roles/base/tasks/packages.yml:30]: Cower doesn't seem to be in the official repos.

File: group_vars/all [hunk 2]
```diff
@@ -60,7 +60,8 @@ base_packages:
     - coreutils
     - moreutils
     - pass
-    - reflector
+    - packer # instead of aura-bin
```
Comment (pigmonkey) [inline group_vars/all:63]: Isn't packer an AUR package?

File: group_vars/all [hunk 3]
```diff
@@ -102,33 +103,34 @@ tarsnapper:
...
 localtime:
-    run_on: trusted
+    run_on: all
 
 nameservers:
     - 8.8.8.8       # Google
     - 209.244.0.3   # Level 3
     - 74.82.42.42   # Hurricane Electric
 
 bluetooth:
```
Comment (pigmonkey) [inline group_vars/all:132]: If we just remove the bluetooth var entirely, nothing in the bluetooth role will be run. Is that preferable to switching `bluetooth.disable` here and then commenting out the rfkill stuff later?

File: roles/base/files/pacman.conf [hunk 4]
```diff
@@ -1,9 +1,78 @@
+#
+# /etc/pacman.conf
```
Comment (pigmonkey) [inline roles/base/files/pacman.conf:2]: If we get rid of all the default comments we could make a cleaner diff between the file here and in master, which may make future maintenance easier.

File: roles/mapping/tasks/main.yml [hunk 5]
```diff
@@ -7,10 +7,10 @@
   tags:
     - aur
 
-- name: Install OpenCPN
-  aur: name=opencpn user={{ user.name }}
-  tags:
-    - aur
+#- name: Install OpenCPN
```
Comment (pigmonkey) [inline roles/mapping/tasks/main.yml:10]: We're being inconsistent between commenting tasks out like this and adding architecture conditionals. I don't have a strong preference between the two methods, but I think keeping it consistent it everywhere will keep our future lives easier.

The conditional would make it easier to merge all of this into master, if down the road we change our minds and decide that is better than keeping the branches separate.

File: roles/rpi/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+---
+- name: Disable and stop haveged
+  service: name=haveged.service enabled=no state=stopped
+  tags:
+    - rpi
```
Comment (pigmonkey) [inline roles/rpi/tasks/main.yml:5]: You tagged the whole with `rpi` in the playbook so the individual task tags here are redundant.

**Comment by jshuping** [comment]
Thanks for the review and sorry for the delay in responding.  I will make some updates and reapply all roles and get back to you soon!

File: roles/base/tasks/packages.yml (hunk 1 above)
Comment (jshuping) [reply roles/base/tasks/packages.yml:30]: ARM folks are doing a little trick here, using [this repo](https://github.com/archlinuxarm/PKGBUILDs) as the basis to build certain ARM-customized packages and [mirror them](http://mirror.archlinuxarm.org/).  A couple of things like cower are there. They are pulled into regular pacman via the `[aur]` and `[alarm]` sections of pacman.conf.

File: group_vars/all (hunk 2 above)
Comment (jshuping) [reply group_vars/all:63]: Same as above for packer, and no reflector... 'mirror.archlinuxarm.org' uses Geo-IP.

File: group_vars/all (hunk 3 above)
Comment (jshuping) [reply group_vars/all:132]: Generally speaking, my r-pi is stationary and only connected to one network. I suppose I could leave things like `localtime` to still run on `trusted` and define my single trusted network. Let me go that route to make it a cleaner diff against master.

For Bluetooth, I was going to experiment with it and get some bt speakers working, and should hopefully get that section of group_vars/all back to matching with master.

File: roles/base/files/pacman.conf (hunk 4 above)
Comment (jshuping) [reply roles/base/files/pacman.conf:2]: Sounds good, I will update.

File: roles/mapping/tasks/main.yml (hunk 5 above)
Comment (jshuping) [reply roles/mapping/tasks/main.yml:10]: Good point.  OpenCPN is one of my primary use cases, and I found various forks and varying compilation arg suggestions.. planning to play around with it before making a 'recommendation' into this role.  Let me update to comment as such.

File: roles/rpi/tasks/main.yml (hunk 6 above)
Comment (jshuping) [reply roles/rpi/tasks/main.yml:5]: thanks, will fix.

**Comment by pigmonkey** [comment]
Stale. Closing.

**Pull request closed without merging**